index.test-d.ts

```diff
@@ -0,0 +1,73 @@
+import fastify from "fastify";
+import fastifyMysql, {
+  MySQLConnection,
+  MySQLPool,
+  MySQLPromiseConnection,
+  MySQLPromisePool,
+} from ".";
+
+const app = fastify();
+app
+  .register(fastifyMysql, {
+    connectionString: "mysql://root@localhost/mysql",
+  })
+  .after(function (err) {
+    const mysql = app.mysql as MySQLPool;
+    mysql.escapeId("foo");
+    mysql.escape("bar");
+    mysql.format("baz");
+    mysql.query("SELECT NOW()", function () {});
+    mysql.execute("SELECT NOW()", function () {});
+    mysql.getConnection(function (err, con) {
+      con.release();
+    });
+    mysql.pool.end();
+  });
+
+app
+  .register(fastifyMysql, {
+    promise: true,
+    connectionString: "mysql://root@localhost/mysql",
+  })
+  .after(async function (err) {
+    const mysql = app.mysql as MySQLPromisePool;
+    mysql.escapeId("foo");
+    mysql.escape("bar");
+    mysql.format("baz");
+    await mysql.query("SELECT NOW()");
+    await mysql.execute("SELECT NOW()");
+    const con = await mysql.getConnection();
+    con.release();
+    mysql.pool.end();
+  });
+
+app
+  .register(fastifyMysql, {
+    type: "connection",
+    connectionString: "mysql://root@localhost/mysql",
+  })
+  .after(async function (err) {
+    const mysql = app.mysql as MySQLConnection;
```

Comment:
Wouldn't it be better to have separate test files that would do type augmentation as recommended per documentation instead of casting?

Reply:
I think the type augmentation in test case will pollute the output of typing's in this package. As we do not ignore the ```test``` directory packed in package.

Reply:
Don't augmentations only work after you import them? Don't think people import tests.

Reply:
Good to know that. I have updated the test case. I need to separate the test into four folder because the type augmentation will affect each and other.

Reply:
Here comes another problem. `tsd` do not support separating type augmentation by `tsconfig.json`. I will use the casting as before.

Reply:
@climba03003 What do you mean by "separating type augmentation by tsconfig.json"? If you simply separate by files, augmentations leak into each other?

Reply:
I think CI failure was caused by wrong TSD configuration, you could do something like this:
```
  "tsd": {
    "directory": "test-tsd",
```

Reply:
Yes, it is. The augmentation affect each and other in my local test. So, I have not updated the test case.

Reply:
got it. oh well.